### PR TITLE
Add ebnf parser

### PR DESCRIPTION
**Pull request by theHamsta**

### CODE REVIEW

**Review by ObserverOfTime — COMMENTED** [review]
I think the `.grammar` suffix is unnecessary since `.ebnf` can always be used.

As for the identifiers, this should be fine:
```
((identifier) @type
 (#match? @type "^[A-Z]"))

((identifier) @symbol
 (#match? @symbol "^[a-z]"))

((identifier) @constant
 (#match? @constant "^[A-Z][A-Z0-9_]+$"))
 ```
 As far as I can tell, the camel/lower distinction is not semantically relevant
 and the fallback capture is redundant as one of the rules will always match.

File: queries/ebnf/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,47 @@
+;;;; Simple tokens ;;;;
+(terminal) @string.grammar
+
+(special_sequence) @string.special.grammar
+
+(integer) @number
+
+(comment) @comment.block
```
[the remark this answers is not in the document]
Comment (theHamsta) [reply queries/ebnf/highlights.scm:8]: would be important to fix CI to allow undocumented subscopes

**Comment by theHamsta** [comment]
@ObserverOfTime please check the original suggestions in the original PR. I just tried to rebase and resolve conflicts (was problematic because old PR was on master with merged branches)

**Comment by theHamsta** [comment]
Should be merged after  #3866

**Review by ObserverOfTime — APPROVED** [review]
I still find the suffix to be unnecessary but it's fine.

**Pull request merged**